contrib/CONTRIBUTING.md

```diff
@@ -0,0 +1,232 @@
+
+
+
+#### Pull Request Philosophy
+
+Patchsets and enhancements should always be focused. For example, a pull request could add a feature, fix a bug, or refactor code; but not a mixture. Please also avoid super pull requests which attempt to do too much, are overly large, or overly complex as this makes review difficult. This helps maintain a separation of concerns and reduces review cognitive load.
+
+In general, all pull requests must:
+
+  - Have a clear use case, fix a demonstrable bug or serve the greater good of the project (for example refactoring for modularisation);
+  - Be well peer-reviewed;
+  - Have unit tests and functional tests, where appropriate;
+  - Follow code style guidelines;
+  - Not break the existing test suite;
+  - Where bugs are fixed, where possible, there should be unit tests demonstrating the bug and also proving the fix. This helps prevent regression.
+  - Change relevant comments and documentation when behaviour of code changes.
+
+#### Pull Request Process
+Please follow these steps to have your contribution considered by the maintainers:
+
+*Before* creating the PR:
+1. Read the [development workflow](./DEVELOPMENT_WORKFLOW.md) defined for this repository in order to agree on the ways of working.
+1. While coding, please, add tests relevant to the fixed bug or new feature. Follow the [testing guide](./TESTING.md)
+1. To create the PR follow all instructions in [the template](https://github.com/opentensor/bittensor/blob/master/.github/PULL_REQUEST_TEMPLATE/pull_request_template.md)
+1. Make sure your commit messages are clear and concise. Include the issue number if applicable.
+1. If you have multiple commits, rebase them into a single commit. Use `git rebase -i` for this.
+1. In the pull request description, explain what the changes do and why you think they should be applied.
+1. If your changes are not ready for merge but you want feedback, create a draft pull request.
+1. After you submit your pull request, verify that all [status checks](https://help.github.com/articles/about-status-checks/) are passing <details><summary>What if the status checks are failing?</summary>If a status check is failing, and you believe that the failure is unrelated to your change, please leave a comment on the pull request explaining why you believe the failure is unrelated. A maintainer will re-run the status check for you. If we conclude that the failure was a false positive, then we will open an issue to track that problem with our status check suite.</details>
```

Comment:
It should not be possible to merge a PR with failing checks. The tests stay failing in the `master`/long running branches in this repository for prolonged periods of time, because these exceptions are abused too much.

Reply:
+1 totally agree, productive-like branches like our master branch should be always operative and in the best state possible. 

If not we are building a [Broken Window](https://en.wikipedia.org/wiki/Broken_windows_theory), and we do not want that.

Reply:
Agreed. Changed wording here to reflect. See: 5e07a8426c7e8d91177fea85138a96295b6269e5

Reply:
For the record there is a repository setting which relates to this and should be enabled:
![image](https://github.com/opentensor/bittensor/assets/36161392/05ccd2e8-ac54-4626-b665-ad2a7e4722ab)
These can be found under the branch protection rules.